src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -454,6 +454,10 @@ end
         # Quick sanity check
         for i = 1:size(X, 2)
             @views if abs(BX[:, i]'X[:, i] - 1) >= sqrt(eps(real(eltype(X))))
+                # TODO error-ing is too harsh here. Better throw some exception (e.g. in
+                # reduced precision this can be interpreted as an indicator to re-try the
+                # orthogonalisation in elevated precision or to stop the reduced precision
+                # iterations as a whole.
                 error("LOBPCG is badly failing to keep the vectors normalized; this should never happen")
```

Comment:
Did this really fail in your tests? X here should be the result of orthogonal operations of matrices returned by ortho, so they really should be orthogonal to some small multiple of machine precision

Reply:
It fails for large-ish systems in Float32. Maybe there is sth to debug, but I could not be bothered.

Reply:
Sent you something.